app/src/main/java/com/example/cinemapp/data/UserPreferences.kt

```diff
@@ -49,4 +61,37 @@ class UserPreferences (context: Context){
         }
     }
 
+    fun getTheme(): Flow<Int?> {
+        return getUserPrefs().map { pref ->
+            pref[appearanceMode]
+        }
+    }
+
+    suspend fun setTheme(mode: Int?) {
+        val newMode = mode ?: getSystemDefaultValue()
+        userPrefs.edit { pref ->
+            pref[appearanceMode] = newMode
+        }
+        setAppTheme(newMode, context)
+    }
+
+
+    fun getLanguage(): Flow<String?> {
```

Comment:
is there a reason why this is a Flow?

Reply:
It's the way it's been done in the documentation and the guides I looked up, but I suppose it can just be a suspend function that returns firstOrNull() of a Flow.